src/Microsoft.IdentityModel.Abstractions/IIdentityLogger.cs

```diff
@@ -0,0 +1,49 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.IdentityModel.Abstractions
+{
+    /// <summary>
+    /// Interface for Logging.
+    /// </summary>
+    public interface IIdentityLogger
```

Comment:
have you looked into leveraging the high performance logging provided by .NET core? We use it in Id Web.
[https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?msclkid=7a0c0097b12c11ecbb637e8c4c4a3a86&view=aspnetcore-6.0](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?msclkid=7a0c0097b12c11ecbb637e8c4c4a3a86&view=aspnetcore-6.0)

Reply:
@jennyf19 I think this is a excellent comment. 
@sruke can you determine if we can add the benefits of high performance logging without any additional changes in a future release.

Reply:
My understanding is that logger extension methods is a traditional logging approach and Logger message provided by .Net Core is more optimized.
* With the logger extension methods all arguments are boxed into objects. LoggerMessage avoids this by using static Action fields.
* With logger extensions methods, the message template is parsed every time a message is written and with LoggerMessage the message template is only parsed once at the time it is defined.

LoggerAdapter that will be written for ASP.Net Core's ILogger will use traditional logger extension methods so changes would be needed.

Moving to .Net Core's high-performance logging would be a significant change for an existing library -
* Wilson would take a direct dependency on Microsoft.Extensions.Logging
* All log messages need to be redefined to follow the below pattern.
```csharp
public static class LoggerExtensions
{
      private static readonly Action<ILogger, string, string, Exception?> debugMessage  =
                LoggerMessage.Define<string, string>(
                    LogLevel.Debug,
                    IDX0001,
                    "TestKey {Arg1} TestValue {Arg2}");

     public static void TestRequest(
                ILogger logger,
                string key,
                string value,
                Exception? ex) => s_l2CacheRetry(
                    logger,
                    cacheType,
                    operation,
                    cacheKey,
                    ex);
}
```

Adopting high performance logging directly would make sense.